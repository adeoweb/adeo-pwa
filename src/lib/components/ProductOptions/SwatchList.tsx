import React, { FunctionComponent, useMemo } from 'react';
import Swatch from './Swatch';
import { TValueListProps } from 'src/lib/components/ProductOptions/ProductOptionsTypes';

const SwatchList: FunctionComponent<TValueListProps> = ({
    getItemKey,
    selectedValue,
    items,
    onSelectionChange
}) => {
    const swatches = useMemo(
        () =>
            items.map(item => {
                const isSelected =
                    Boolean(selectedValue.label) &&
                    item.label === selectedValue.label;

                return (
                    <Swatch
                        key={getItemKey(item)}
                        isSelected={isSelected}
                        item={item}
                        onClick={onSelectionChange}
                    />
                );
            }),
        [getItemKey, selectedValue.label, items, onSelectionChange]
    );

    return <ul className="config-swatch-list">{swatches}</ul>;
};

SwatchList.displayName = 'SwatchList';

export default SwatchList;
