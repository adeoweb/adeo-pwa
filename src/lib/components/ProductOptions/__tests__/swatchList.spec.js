import React from 'react';
import testRenderer from 'react-test-renderer';

import SwatchList from '../SwatchList';

const defaultProps = {
    items: [
        {
            id: '1',
            value_index: 1,
            label: 'foo'
        }
    ],
    selectedValue: {},
    getItemKey: jest.fn(),
    onSelectionChange: jest.fn()
};

test('renders SwatchList component correctly', () => {
    const component = testRenderer.create(<SwatchList {...defaultProps} />);
    expect(component.toJSON()).toMatchSnapshot();
});
