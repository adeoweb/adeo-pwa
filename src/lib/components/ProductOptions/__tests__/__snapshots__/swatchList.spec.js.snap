// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SwatchList component correctly 1`] = `
<ul
  className="config-swatch-list"
>
  <li
    className=""
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "foo",
        }
      }
      title="foo"
    />
  </li>
</ul>
`;
